modules/markup/external/external.go

```diff
@@ -98,7 +98,14 @@ func (p *Renderer) Render(ctx *markup.RenderContext, input io.Reader, output io.
 		args = append(args, f.Name())
 	}
 
-	processCtx, cancel := context.WithCancel(ctx.Ctx)
+	parentCtx := context.Background()
+	if ctx != nil && ctx.Ctx != nil {
+		parentCtx = ctx.Ctx
+	} else {
+		log.Error("RenderContext did not provide context)")
+	}
+
+	processCtx, cancel := context.WithCancel(parentCtx)
```

Comment:
Not setting the renderctx is a fatal error and should not be tolerated.

Reply:
ok will return an error then :)